app/services/FilterService.scala

```diff
@@ -0,0 +1,139 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import config.AppConfig
+import models.getPenaltyDetails.GetPenaltyDetails
+import models.getPenaltyDetails.appealInfo.AppealStatusEnum
+import models.getPenaltyDetails.latePayment.{LPPDetails, LPPPenaltyCategoryEnum, LPPPenaltyStatusEnum, LatePaymentPenalty}
+import models.getPenaltyDetails.lateSubmission.{LSPDetails, LateSubmissionPenalty}
+import play.api.libs.json.{JsString, JsValue, Json}
+import utils.Logger.logger
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+import scala.util.Try
+
+class FilterService @Inject()()(implicit ec: ExecutionContext, appConfig: AppConfig) {
+
+  def tryJsonParseOrJsSting(body: String): JsValue = {
+    Try(Json.parse(body)).getOrElse(JsString(body))
+  }
+
+  def filterEstimatedLPP1DuringPeriodOfFamiliarisation(penaltiesDetails: GetPenaltyDetails, callingClass: String, function: String, vrn: String): GetPenaltyDetails = {
+    if (penaltiesDetails.latePaymentPenalty.nonEmpty) {
+      val filtered: Option[Seq[LPPDetails]] = filterEstimatedLPP1(penaltiesDetails)
+      val numberOfFiltered = numberOfFilteredLPPs(filtered, penaltiesDetails)
+      if (filtered.nonEmpty && filtered.get.nonEmpty && numberOfFiltered >= 0) {
+        logger.info(s"[FilterService][filterEstimatedLPP1DuringPeriodOfFamiliarisation] Filtering for [$callingClass][$function] -" +
+          s" Filtered ${numberOfFiltered} LPP1(s) from payload for VRN: $vrn")
+        penaltiesDetails.copy(latePaymentPenalty = Some(LatePaymentPenalty(filtered)))
+      } else {
+        logger.info(s"[FilterService][filterEstimatedLPP1DuringPeriodOfFamiliarisation] Filtering for [$callingClass][$function] - Filtered all LPP1s from payload for VRN: $vrn")
+        penaltiesDetails.copy(latePaymentPenalty = None)
+      }
+    } else {
+      logger.info(s"[FilterService][filterEstimatedLPP1DuringPeriodOfFamiliarisation] Filtering for [$callingClass][$function] - No LPPs to filter for VRN: $vrn")
+      penaltiesDetails
+    }
+  }
+
+  private def filterEstimatedLPP1(penaltiesDetails: GetPenaltyDetails): Option[Seq[LPPDetails]] = {
+    penaltiesDetails.latePaymentPenalty.flatMap(
+      _.details.map(latePaymentPenalties => latePaymentPenalties.filterNot(lpp => {
+        lpp.penaltyCategory.equals(LPPPenaltyCategoryEnum.FirstPenalty) &&
+          lpp.penaltyStatus.equals(LPPPenaltyStatusEnum.Accruing) &&
+          appConfig.withinLPP1FilterWindow(lpp.principalChargeDueDate)
+      })
+      )
+    )
+  }
+
+  def filterPenaltiesWith9xAppealStatus(penaltiesDetails: GetPenaltyDetails)(implicit callingClass: String, function: String, vrn: String): GetPenaltyDetails = {
+    val filteredLSPs: Option[Seq[LSPDetails]] = if (penaltiesDetails.lateSubmissionPenalty.nonEmpty) filterLSPWith9xAppealStatus(penaltiesDetails) else None
```

Comment:
Why use if? Could we not just call the filterLSPWith9xAppealStatus func directly?